### PR TITLE
Webaudio workaround (Broadcom) [memento]

### PR DESCRIPTION
**Pull request by igor-borovkov**
During navigation on youtube.com/tv, sound is not played

1. Broadcom gst plugins do not provide buffers with decoded data.
   Dummy buffers are pushed.
2. Audio is played during decoding and not by audiosink.

Changed pipeline to
giostreamsrc ! mpegaudioparse ! brcmmp3decoder ! audioconvert
! audioresample ! capsfilter ! deinterleave ! queue ! appsink

### CODE REVIEW

File: Source/WebCore/platform/audio/gstreamer/AudioFileReaderGStreamer.cpp [hunk 1]
```diff
@@ -340,7 +345,17 @@ void AudioFileReader::decodeAudioForBusCreation()
...
+#if PLATFORM(BROADCOM)
+    GstElement* parse = gst_element_factory_make("mpegaudioparse", NULL);
+    m_decodebin = gst_element_factory_make("brcmmp3decoder", NULL);
+
+    gst_bin_add_many(GST_BIN(m_pipeline.get()), source, parse, WTF::refGPtr(m_decodebin.get()), NULL);
 
+    gst_element_link_pads_full(source, "src", parse, "sink", GST_PAD_LINK_CHECK_NOTHING);
+    gst_element_link_pads_full(parse, "src", m_decodebin.get(), "brcm-mp3-sink", GST_PAD_LINK_CHECK_NOTHING);
+
+    plugDeinterleave(nullptr);
```
Comment (charlie-ht) [inline Source/WebCore/platform/audio/gstreamer/AudioFileReaderGStreamer.cpp:357]: Can we not use "pad-added" callback on the `brcmmp3decoder` element to call this method like we do on the normal decodebin?

Where's the documentation for the BROADCOM gstreamer element in use out of interest?

**Review by charlie-ht — COMMENTED** [review]
I'll add @eocanha for a review since he knows more about Broadcom, in general I wonder if we could one day upstream this, are the broadcom elements available publically and their documentation?

As a general comment, I wonder if we need a separate AudioFileReaderGStreamerBroadcom at this point :-)

**Comment by woutermeek** [comment]
@igor-borovkov I guess this is not needed anymore since you are moving to 2.22?

**Comment by woutermeek** [comment]
Closing, please re-open if still needed

**Pull request closed without merging**